pyvrp/accept/MovingAverageThreshold.py

```diff
@@ -0,0 +1,151 @@
+
+    where :math:`w_0 \ge 0` is the initial weight parameter, :math:`T \ge 0`
+    and :math:`I \ge 0` are the maximum runtime and iterations parameters,
+    and :math:`t` and :math:`i` are the elapsed runtime and number of
+    iterations. The dynamic weight uses whichever limit (runtime or iterations)
+    is most restrictive.
+
+    Note: The parameters ``weight`` and ``history_length`` correspond to
+    :math:`\eta` and :math:`\gamma` respectively in [1]_.
+
+    Parameters
+    ----------
+    weight
+        Initial weight parameter :math:`w_0` used to determine the threshold
+        value. Larger values result in more accepted candidate solutions. Must
+        be non-negative.
+    history_length
+        The number of recent candidate solutions to consider when computing
+        the threshold value. Must be positive.
+    max_runtime
+        Maximum runtime in seconds. As the search approaches this time limit,
+        :math:`w \to 0`. Must be non-negative. Default is ``None``, meaning
+        that :math:`w` stays equal to :math:`w_0`.
+    max_iterations
+        Maximum number of iterations. As the search approaches this limit,
+        :math:`w \to 0`. Must be non-negative. Default is ``None``, meaning
+        that :math:`w` stays equal to :math:`w_0`.
+
+    References
+    ----------
+    .. [1] Máximo, V.R. and M.C.V. Nascimento. 2021. A hybrid adaptive iterated
+           local search with diversification control to the capacitated vehicle
+           routing problem, *European Journal of Operational Research* 294 (3):
+           1108 - 1119.
+    """
+
+    def __init__(
+        self,
+        weight: float,
+        history_length: int,
+        max_runtime: float | None = None,
+        max_iterations: int | None = None,
+    ):
+        if weight < 0:
+            raise ValueError("weight must be non-negative.")
+
+        if history_length <= 0:
+            raise ValueError("history_length must be positive.")
+
+        if max_runtime is not None and max_runtime < 0:
+            raise ValueError("max_runtime must be non-negative.")
+
+        if max_iterations is not None and max_iterations < 0:
+            raise ValueError("max_iterations must be non-negative.")
+
+        self._weight = weight
+        self._history_length = history_length
+        self._max_runtime = max_runtime
+        self._max_iterations = max_iterations
+
+        self._history = np.zeros(history_length)
+        self._start_time = perf_counter()
+        self._iters = 0
+
+    @property
+    def weight(self) -> float:
+        return self._weight
+
+    @property
+    def history_length(self) -> int:
+        return self._history_length
+
+    @property
+    def max_runtime(self) -> float | None:
+        return self._max_runtime
+
+    @property
+    def max_iterations(self) -> float | None:
+        return self._max_iterations
+
+    @property
+    def _runtime_budget(self) -> float:
```

Comment:
There's no fixed definition for what is and what is not a property, but my heuristic is:
- It should be constant-ish. If I repeatedly access a property without changing anything else, I should get the same result.
- Accessing a property should not do meaningful work. It's basically equivalent to looking up some data on an object.

This property somewhat violates both those 'rules'. Perhaps it should just be a regular method. For good measure, we should then probably also update `_iteration_budget` to match.